tests/components/airpatrol/test_config_flow.py

```diff
@@ -0,0 +1,319 @@
+"""Test the AirPatrol config flow."""
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from airpatrol.api import AirPatrolAuthenticationError, AirPatrolError
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.airpatrol.const import DOMAIN
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+TEST_USER_INPUT = {
+    CONF_EMAIL: "test@example.com",
+    CONF_PASSWORD: "test_password",
+}
+
+MOCK_AUTH_RESPONSE = {
+    "status": "ok",
+    "entities": {
+        "users": {
+            "list": [
+                {
+                    "appId": "test_unique_id",
+                }
+            ]
+        }
+    },
+    "misc": {"accessToken": "test_access_token"},
+}
+
+
+@pytest.fixture
+def mock_api():
+    """Mock AirPatrol API."""
+    api = MagicMock()
+    api.get_unique_id.return_value = "test_user_id"
+    api.get_access_token.return_value = "test_access_token"
+    return api
+
+
+@pytest.fixture
+def mock_config_entry():
+    """Mock config entry."""
+    return MockConfigEntry(
+        domain="airpatrol",
+        data={
+            "email": "test@example.com",
+            "password": "test_password",
+            "access_token": "test_access_token",
+        },
+        unique_id="test_user_id",
+    )
+
+
+@pytest.mark.asyncio
+async def test_async_step_reauth_confirm_success(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test successful reauthentication via async_step_reauth_confirm."""
+    mock_config_entry.add_to_hass(hass)
+    flow = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "reauth", "entry_id": mock_config_entry.entry_id}
+    )
+    # Should show the reauth_confirm form
+    assert flow["type"] == "form"
+    assert flow["step_id"] == "reauth_confirm"
+
+    with patch(
+        "homeassistant.components.airpatrol.config_flow.async_get_clientsession"
+    ) as mock_session:
+        mock_session.return_value = AsyncMock()
+        with patch(
+            "homeassistant.components.airpatrol.config_flow.AirPatrolAPI.authenticate",
+            new_callable=AsyncMock,
+        ) as mock_auth:
```

Comment:
Mock the whole `AirPatrolAPI` in `conftest` with it's responses and then use it as a fixture in the tests instead of patching in every test.